src/ripple/test/impl/WSClient.cpp

```diff
@@ -107,6 +107,15 @@ class WSClientImpl : public WSClient
     std::condition_variable cv_;
     std::list<std::shared_ptr<msg>> msgs_;
 
+    void cleanup()
+    {
+        error_code ec;
+        ws_.close({}, ec);
+        stream_.close(ec);
+        work_ = boost::none;
```

Comment:
oops - turns out it serves a purpose. `work_` is constructed by `WSClientImpl`, and according to the docs:
```
  /**
   * The constructor is used to inform the io_service that some work has begun.
   * This ensures that the io_service object's run() function will not exit
   * while the work is underway.
   */
```
Removing `work_` resulted in significant test failures for me. Looks like we need it.